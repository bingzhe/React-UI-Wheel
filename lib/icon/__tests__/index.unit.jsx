import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from "../index";
import { mount } from 'enzyme';

describe('icon', () => {
    it('render alipay icon.', () => {
        const json = renderer.create(<Icon name="alipay" />).toJSON();
        expect(json).toMatchSnapshot();
    });
    it('render wechat icon.', () => {
        const json = renderer.create(<Icon name="wechat" />).toJSON();
        expect(json).toMatchSnapshot();
    });
    it('onClick', () => {
        const fn = jest.fn();
        const component = mount(<Icon name="alipay" onClick={fn} />);
        component.find('svg').simulate('click');
        expect(fn).toBeCalled();
    });
});