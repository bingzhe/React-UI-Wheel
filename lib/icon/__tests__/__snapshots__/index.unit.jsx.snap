// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render success 1`] = `
<svg
  className="r-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
